### ISSUE TITLE
-limit Parameter required on Invoke-ArubaCPRestMethod

### ISSUE THREAD

**Issue by cmcknz77** [PowerAruba/PowerArubaCP#14]
I have many more than 25 NAD objects in my CPPM NAD list and by default Invoke-ArubaCPRestMethod only pulls the first 25 objects which means that the NetworkDevices functions cannot find any objects by name that are not within the first 25 that an 'Invoke-ArubaCPRestMethod' -uri "api/network-device" -method 'GET' command would list...

So, we need to find a way to pass the appropriate 'limit' and 'filter' options into Invoke-ArubaCPRestMethod and amend Get-ArubaCPNetworkDevice accordingly.

As an example, I have the same number of network objects in FRPAR and UKLON (3 each).  The FRPAR objects are in the first 25, the UKLON devices are not and as a result I get the following results when trying to search for them:

```powershell
PS C:\> Get-ArubaCPNetworkDevice -Name FRPAR
id             : 3006
name           : EU-FRPAR-VC1
description    : Aruba Virtual Controller - Paris
ip_address     : 192.168.32.100
radius_secret  :
tacacs_secret  :
vendor_name    : Aruba
coa_capable    : True
coa_port       : 3799
radsec_enabled : False
snmp_read      : @{snmp_version=V2C; community_string=; force_read=False; read_arp_info=True}
snmp_write     : @{snmp_version=V2C; community_string=}
attributes     :
_links         : @{self=}

id             : 3019
name           : FRPAR-Switches
description    : FRPAR-Switches
ip_address     : 192.168.32.1/29
radius_secret  :
tacacs_secret  :
vendor_name    : Aruba
coa_capable    : True
coa_port       : 3799
radsec_enabled : False
snmp_read      : @{snmp_version=V2C; community_string=; force_read=True; read_arp_info=True; zone_name=default}
attributes     :
_links         : @{self=}

id             : 3026
name           : FRPAR-Core
description    : FRPAR-Switches
ip_address     : 192.168.216.254
radius_secret  :
tacacs_secret  :
vendor_name    : Aruba
coa_capable    : True
coa_port       : 3799
radsec_enabled : False
snmp_read      : @{snmp_version=V2C; community_string=; force_read=True; read_arp_info=True; zone_name=default}
attributes     :
_links         : @{self=}

PS C:\> Get-ArubaCPNetworkDevice -Name UKLON
PS C:\>
```

**Comment by alagoutte** [comment]
Thanks, i will look

for PowerArubaCX, i have already for Invoke-(ArubaCP)RestMethod, the possibility to pass some parameter

I will make the same for this module

and also make the filter by ClearPass/API and not by powershell (with Where-Object)

**Comment by alagoutte** [comment]
Sorry for delay but busy

but i will add limit paramter and better filter stuff

**Comment by alagoutte** [comment]
Limit parameter is now available for Invoke-ArubaCPRestMethod (and also Get-ArubaCPNetworkDevice)

but it is limited to 1000 and i add filter option for Invoke-ArubaCPRestmethod in #20 and for NetworkDevice in #21

**Comment by alagoutte** [comment]
Hi,

a new release is available (0.3.0) on [PS Gallery ](https://www.powershellgallery.com/packages/PowerArubaCP/0.3.0)with some fix for this issue
